plugins/file_chooser/gtk/file_chooser_plugin.cc

```diff
@@ -0,0 +1,201 @@
+                                     GtkFileChooserAction action,
+                                     const gchar* title,
+                                     const gchar* default_confirm_button_text,
+                                     FlValue* properties) {
+  if (fl_value_get_type(properties) != FL_VALUE_TYPE_MAP) {
+    return FL_METHOD_RESPONSE(fl_method_error_response_new(
+        "Bad Arguments", "Argument map missing or malformed", nullptr));
+  }
+
+  const gchar* confirm_button_text = default_confirm_button_text;
+  FlValue* value = fl_value_lookup_string(properties, kConfirmButtonTextKey);
+  if (value != nullptr && fl_value_get_type(value) == FL_VALUE_TYPE_STRING)
+    confirm_button_text = fl_value_get_string(value);
+
+  FlView* view = fl_plugin_registrar_get_view(self->registrar);
+  GtkWindow* window = GTK_WINDOW(gtk_widget_get_parent(GTK_WIDGET(view)));
+  g_autoptr(GtkFileChooserNative) dialog =
+      GTK_FILE_CHOOSER_NATIVE(gtk_file_chooser_native_new(
+          title, window, action, confirm_button_text, "_Cancel"));
+
+  value = fl_value_lookup_string(properties, kAllowsMultipleSelectionKey);
+  if (value != nullptr && fl_value_get_type(value) == FL_VALUE_TYPE_BOOL) {
+    gtk_file_chooser_set_select_multiple(GTK_FILE_CHOOSER(dialog),
+                                         fl_value_get_bool(value));
+  }
+
+  value = fl_value_lookup_string(properties, kCanChooseDirectoriesKey);
+  if (value != nullptr && fl_value_get_type(value) == FL_VALUE_TYPE_BOOL &&
+      fl_value_get_bool(value)) {
+    gtk_file_chooser_set_action(GTK_FILE_CHOOSER(dialog),
+                                GTK_FILE_CHOOSER_ACTION_SELECT_FOLDER);
+  }
+
+  value = fl_value_lookup_string(properties, kInitialDirectoryKey);
+  if (value != nullptr && fl_value_get_type(value) == FL_VALUE_TYPE_STRING) {
+    gtk_file_chooser_set_current_folder(GTK_FILE_CHOOSER(dialog),
+                                        fl_value_get_string(value));
+  }
+
+  value = fl_value_lookup_string(properties, kInitialFileNameKey);
+  if (value != nullptr && fl_value_get_type(value) == FL_VALUE_TYPE_STRING) {
+    gtk_file_chooser_set_current_name(GTK_FILE_CHOOSER(dialog),
+                                      fl_value_get_string(value));
+  }
+
+  value = fl_value_lookup_string(properties, kAllowedFileTypesKey);
+  if (value != nullptr && fl_value_get_type(value) == FL_VALUE_TYPE_LIST) {
+    for (size_t i = 0; i < fl_value_get_length(value); i++) {
+      FlValue* file_type = fl_value_get_list_value(value, i);
+      g_autoptr(GtkFileFilter) filter = file_type_to_filter(file_type);
+      if (filter == nullptr) {
+        return FL_METHOD_RESPONSE(fl_method_error_response_new(
+            "Bad Arguments", "Allowed file types malformed", nullptr));
+      }
+      gtk_file_chooser_add_filter(GTK_FILE_CHOOSER(dialog), filter);
+    }
+  }
+
+  gint response = gtk_native_dialog_run(GTK_NATIVE_DIALOG(dialog));
+  g_autoptr(FlValue) result = fl_value_new_list();
```

Comment:
Cancel needs to be null, not an empty list. Fixing now.

Reply:
Ah, I didn't notice that CreateResponseObject() in the old Linux plugin made a null value on an empty list.